### ISSUE TITLE
Elsa cannot publish IO list data

### ISSUE THREAD

**Issue by davydog187** [bbalser/elsa#77]
We are Avro encoding data, and it appears that Elsa is not handling Avro encoded data in IO list form.

```
** (ArgumentError) argument error
:erlang.byte_size([<<0, 0, 0, 0, 3>>, [[2], [[[[0], ['H', \"c21e6bf8-9a1b-44a2-b3f9-7638843c488c\"]], [[2], \"\"], [[2], \"\"]]]]])
(elsa 0.12.2) lib/elsa/util.ex:182: Elsa.Util.get_byte_size/1
(elsa 0.12.2) lib/elsa/util.ex:151: Elsa.Util.chunk/4
(elixir 1.10.1) lib/enum.ex:511: anonymous fn/3 in Enum.chunk_while/4
(elixir 1.10.1) lib/enum.ex:3686: Enumerable.List.reduce/3
(elixir 1.10.1) lib/enum.ex:510: Enum.chunk_while/4
(elsa 0.12.2) lib/elsa/producer.ex:111: anonymous fn/1 in Elsa.Producer.create_message_chunks/2
(elixir 1.10.1) lib/enum.ex:1400: anonymous fn/3 in Enum.map/2
(stdlib 3.11.2) maps.erl:232: :maps.fold_1/3
(elixir 1.10.1) lib/enum.ex:2127: Enum.map/2
(elsa 0.12.2) lib/elsa/producer.ex:111: Elsa.Producer.create_message_chunks/2
(elsa 0.12.2) lib/elsa/producer.ex:78: anonymous fn/4 in Elsa.Producer.do_produce_sync/4
(elsa 0.12.2) lib/elsa/producer.ex:65: Elsa.Producer.ad_hoc_produce/5
(elsa 0.12.2) lib/elsa/producer.ex:37: Elsa.Producer.produce/4
```

**Comment by davydog187** [comment]
The fix on our end is to call https://hexdocs.pm/elixir/IO.html#iodata_to_binary/1 before encoding, but really Elsa should be able to handle IO lists gracefully. There shouldn't be need to convert to binary before the wire.

**Comment by jeffgrunewald** [comment]
You're not wrong. The tricky part is going to be chunking the data to produce since we've had better luck doing so by size in bytes and the kernel `byte_size/1` doesn't support charlists.

Looking at the code in the Utils module, it looks like we originally intended to allow the user to override the default chunking function which would be the easiest way to solve this problem in once sense while at the same time making it a "you" problem to figure out the chunking. Since I don't _love_ that approach, I'll take a look and see if I can think of an alternative to chunking when the message is made of iodata rather than binary data.

**Comment by davydog187** [comment]
The problem with the current implementation is that is a surprising interface, and fails in an unexpected way

**Comment by jeffgrunewald** [comment]
https://github.com/bbalser/elsa/pull/84 should address this.

**Comment by jdenen** [comment]
This has been released as part of `v1.0.0-rc.1`.
